app/src/main/java/com/eveningoutpost/dexdrip/Models/BgReading.java

```diff
@@ -678,6 +682,70 @@ public String noiseValue() {
             return String.valueOf(noise);
         }
     }
+    
+    /*
+     * This function comes to check weather we are in a case that we have an allert but since things are
+     * getting better we should not do anything. (This is only in the case that the alert was snoozed before.)
+     * This means that if this is a low alert, and we have two readings in the last 15 minutes, and
+     * either we have gone in 10 in the last two readings, or we have gone in 3 in the last reading, we
+     * don't play the alert again, but rather wait for the alert to finish.
+     *  I'll start with having the same values for the high alerts.
+    */ 
+    
+    public static boolean trendingToAlertEnd(Context context, boolean above) {
```

Comment:
I think this is a great idea, its going to be tough to explain it, but its going to save so many unnecessary alerts from waking people up!